### PR TITLE
Fix schema validation errors

### PR DESCRIPTION
**Pull request by alemorcuq**
**Description of the change**
Fix issues reported in #955 regarding schema validations:

- Allow `spec.template.data` to be `null`
- Preserve unknown fields for `spec.template.metadata` to maintain labels and avoid null validations such as `createTimestamp`

Also:

- Use the schema in the `controller.yaml` files

**Applicable issues**
- #955

### CODE REVIEW

**Review by alvneiayu — DISMISSED** [review]
LGTM

File: schema-v1alpha1.yaml [hunk 1]
```diff
@@ -1,8 +1,98 @@
 openAPIV3Schema:
-  type: object
+  description: SealedSecret is the K8s representation of a "sealed Secret" -
+    a regular k8s Secret that has been sealed (encrypted) using the controller's
```
Comment (josvazg) [inline schema-v1alpha1.yaml:3]: How can we omit `object` here? and why?

File: Makefile [hunk 2]
```diff
@@ -55,7 +55,6 @@ generate: $(GO_FILES)
 	cp -r gentmp/github.com/bitnami-labs/sealed-secrets/pkg . && rm gentmp/ -rf
 
 manifests:
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0)
 	$(CONTROLLER_GEN) crd paths="./pkg/apis/..."  output:crd:artifacts:config=helm/sealed-secrets/crds/
```
Comment (josvazg) [inline Makefile:58]: Why do we need to remove "go getting" the controller-gen binary?

File: helm/sealed-secrets/crds/bitnami.com_sealedsecrets.yaml [hunk 3]
```diff
@@ -118,9 +120,3 @@ spec:
         type: object
     served: true
     storage: true
-status:
-  acceptedNames:
-    kind: ""
-    plural: ""
-  conditions: []
-  storedVersions: []
```
Comment (josvazg) [inline helm/sealed-secrets/crds/bitnami.com_sealedsecrets.yaml:126]: Is this required?

File: schema-v1alpha1.yaml (hunk 1 above)
Comment (alemorcuq) [reply schema-v1alpha1.yaml:3]: It was a typo at copy&pasting. Should be there now.

File: Makefile (hunk 2 above)
Comment (alemorcuq) [reply Makefile:58]: That is not defined and therefore doesn't work. Apart from that, we are not downloading any binaries on the Makefile.

File: helm/sealed-secrets/crds/bitnami.com_sealedsecrets.yaml (hunk 3 above)
Comment (alemorcuq) [reply helm/sealed-secrets/crds/bitnami.com_sealedsecrets.yaml:126]: This was removed by `controller-gen` version `0.9.2`.

File: Makefile (hunk 2 above)
Comment (josvazg) [reply Makefile:58]: Fair enough

**Review by josvazg — APPROVED** [review]
LGTM

**Review by alvneiayu — APPROVED** [review]
LGTM

**Comment by brokenjacobs** [comment]
LGTM validation issues seem to be gone.

**Pull request merged**